lnwallet/wallet.go

```diff
@@ -2543,7 +2549,33 @@ func (c *CoinSource) ListCoins(minConfs int32,
 	}
 
 	var coins []wallet.Coin
+
+	// sweepLabel is the label the sweeper subsystem uses to mark
+	// transactions which were created and published by it.
+	sweepLabel := labels.MakeLabel(
+		labels.LabelTypeSweepTransaction, nil,
+	)
+
 	for _, utxo := range utxos {
+		// We don't include utxos which are unconfirmed yet and
+		// resulted from the sweeper subsystem, because those
+		// unconfirmed utxos can still be RBFed therefore causing side
+		// effects especially when opening zeroconf channels.
+		if utxo.Confirmations == 0 {
+			label, err := c.wallet.FetchTxLabel(utxo.Hash)
+			if err != nil {
+				return nil, err
+			}
+
+			if strings.Contains(label, sweepLabel) {
```

Comment:
Maybe this check makes the `ListCoins` function to specific and we should probably aim for something more generic ?  Maybe allow a list of labels as an input to `ListCoins` so that we can be more flexible in the future ?

Reply:
I feel hesitant to use the labels for this, but I'm not deeply familiar with the lnwallet code. What would be the problem with disallowing *all* unconfirmed utxos irrespective of labels?

Reply:
I thought about that as well which we could enforce by just not allowing to setting the `minConfs` parameter in the `OpenChannelRequest`, tho having the ability to open channels with unconfirmed outputs makes acutally a lot of sense. Imagine a LSP opening a zero-conf channel, he knows that this transaction will not be replaced otherwise he would not open the zero-conf in the first place. If the LSP does only have one big utxo, he will not be able to use the change output of the channel opening to open a potential new zero-conf channel for another user. So I think we should definitely support using unconfirmed utxos for channel openings. But this is a very specific scenario tailored for LSPs.

Reply:
Please what is LSP?

Reply:
I guess, Lightning service provider, never mind.